plenum/test/replica_removing/test_replica_removing.py

```diff
@@ -102,22 +102,31 @@ def test_ordered_request_freed_on_replica_removal(looper,
                                                   chkFreqPatched,
                                                   view_change):
     node = txnPoolNodeSet[0]
-    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
+    if node.master_replica._consensus_data.checkpoints:
+        num_requests_to_stable_checkpoint = node.master_replica._consensus_data.stable_checkpoint - \
+                                            node.master_replica._consensus_data.checkpoints[-1].seqNoEnd
+        sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client,
+                                  num_requests_to_stable_checkpoint)
+    assert not node.master_replica._consensus_data.checkpoints
+    old_stable_checkpoint = node.master_replica._consensus_data.stable_checkpoint
+    assert len(node.requests) == 0
+    with delay_rules(node.nodeIbStasher, cDelay()):
+        sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
 
-    assert len(node.requests) == 1
+        assert len(node.requests) == 1
 
-    forwardedToBefore = next(iter(node.requests.values())).forwardedTo
-    node.replicas.remove_replica(node.replicas.num_replicas - 1)
+        forwardedToBefore = next(iter(node.requests.values())).forwardedTo
+        node.replicas.remove_replica(node.replicas.num_replicas - 1)
 
-    assert len(node.requests) == 1
-    forwardedToAfter = next(iter(node.requests.values())).forwardedTo
-    assert forwardedToAfter == forwardedToBefore - 1
-    check_for_nodes(txnPoolNodeSet, check_stable_checkpoint, 0)
+        assert len(node.requests) == 1
+        forwardedToAfter = next(iter(node.requests.values())).forwardedTo
+        assert forwardedToAfter == forwardedToBefore - 1
+    looper.run(eventually(check_for_nodes, txnPoolNodeSet, check_stable_checkpoint, old_stable_checkpoint))
 
     # Send one more request to stabilize checkpoint
     sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
-
-    looper.run(eventually(check_for_nodes, txnPoolNodeSet, check_stable_checkpoint, 3))
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
```

Comment:
Is it a duplicate line?

Reply:
We send 2 batches. But it will be better to use the `sdk_send_batches_of_random_and_check()`.